torch_frame/data/dataset.py

```diff
@@ -112,53 +140,44 @@ class DataFrameToTensorFrameConverter:
-            specifying :obj:`text_tokenizer` that maps sentences into a
-            list of dictionary of tensors. Each element in the list
-            corresponds to each sentence, keys are input arguments to
-            the model such as :obj:`input_ids`, and values are tensors
-            such as tokens. :obj:`batch_size` specifies the mini-batch
-            size for :obj:`text_tokenizer`. (default: :obj:`None`)
-        col_to_time_format (Union[str, Dict[str, str]], optional): A
-            dictionary or a string specifying the format for the timestamp
-            columns. See `strfttime documentation
+        col_to_sep (Dict[str, Optional[str]]): A dictionary specifying the
+            separator/delimiter for the multi-categorical columns.
+            (default: :obj:`{}`)
+        col_to_text_embedder_cfg (Dict[str, TextEmbedderConfig]):
+            A dictionary of configurations specifying :obj:`text_embedder` that
+            maps text columns into :class:`torch.nn.Embeddings` and
+            :obj:`batch_size` that specifies the mini-batch size for
+            :obj:`text_embedder`. (default: :obj:`{}`)
+        col_to_text_tokenizer_cfg (Dict[str, TextTokenizerConfig]):
+            A dictionary of text tokenizer configurations, specifying
+            :obj:`text_tokenizer` that maps sentences into a list of dictionary
+            of tensors. Each element in the list corresponds to each sentence,
+            keys are input arguments to the model such as :obj:`input_ids`, and
+            values are tensors such as tokens. :obj:`batch_size` specifies the
+            mini-batch size for :obj:`text_tokenizer`. (default: :obj:`{}`)
+        col_to_time_format (Dict[str, Optional[str]]): A dictionary of the
+            time format for the timestamp columns. See `strfttime documentation
             <https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior>`_
             for more information on formats. If a string is specified,
             then the same format will be used throughout all the timestamp
             columns. If a dictionary is given, we use a different format
             specified for each column. If not specified, Pandas' internal
             to_datetime function will be used to auto parse time columns.
-            (default: None)
+            (default: :obj:`{}`)
     """
     def __init__(
         self,
         col_to_stype: dict[str, torch_frame.stype],
         col_stats: dict[str, dict[StatType, Any]],
         target_col: str | None = None,
-        col_to_sep: str | dict[str, str] = ",",
-        col_to_text_embedder_cfg: None |
-        (dict[str, TextEmbedderConfig] | TextEmbedderConfig) = None,
-        col_to_text_tokenizer_cfg: None |
-        (dict[str, TextTokenizerConfig] | TextTokenizerConfig) = None,
-        col_to_time_format: str | dict[str, str] | None = None,
+        col_to_sep: dict[str, str | None] = {},
```

Comment:
See this https://chat.openai.com/share/3506c8b0-4810-42f2-a23a-a9d4e3c38fe4

Reply:
I deliberately made so, since the input here is supposed to be canonicalized dictionary. By default, `col_to_pattern` should be an empty dictionary.

Reply:
Updated. Since empty dictionary will be never accessed, leaving them as `None` inside the class seems fine as well. Again, thanks for the good catch!